mods/persistence/modules/world_save/serializers/json_serializer.dm

```diff
@@ -77,7 +79,12 @@
 /serializer/json/DeserializeDatum(var/datum/persistence/load_cache/thing/object)
 	throw EXCEPTION("Do not use DeserializeDatum for the JSON Serializer. Use QueryAndDeserializeDatum.")
 
-/serializer/proc/JsonDeserializeDatum(var/datum/thing, var/list/thing_vars)
+/serializer/proc/JsonDeserializeDatum(var/thing_type, var/list/thing_vars, var/thing_id)
+	var/datum/thing = reverse_map["[thing_id]"]
```

Comment:
Why is a memory address being assigned in reverse_map? The key for reverse_map is always the index of the table row for that object.